src/cli_util.rs

```diff
@@ -1386,15 +1386,25 @@ fn load_revset_aliases(
 pub fn resolve_base_revs(
     workspace_command: &WorkspaceCommandHelper,
     revisions: &[RevisionArg],
+    multi: bool,
 ) -> Result<IndexSet<Commit>, CommandError> {
     let mut commits = IndexSet::new();
-    for revision_str in revisions {
-        let commit = workspace_command.resolve_single_rev(revision_str)?;
-        let commit_hash = short_commit_hash(commit.id());
-        if !commits.insert(commit) {
-            return Err(user_error(format!(
-                r#"More than one revset resolved to revision {commit_hash}"#,
-            )));
+    if multi {
+        for revset in revisions {
+            let revisions = workspace_command.resolve_revset(revset)?;
+            workspace_command.check_non_empty(&revisions)?;
+            // It's OK if one of the base revs includes a non-rewritable commit
```

Comment:
`resolve_destination_revs`? It needs slightly different logic to work with `-s` and the rest, since that would need to check that revisions are rewriteable.

Reply:
Maybe it's better to have a separate function for checking that a set of commits are rewritable, assuming we don't need the revset expression as context for the error message? Actually, if we moved that out of `resolve_multiple_rewritable_revsets()`, we could reuse the remainder of that function here, right?

Reply:
I refactored and moved around these functions a bit. I hope that now it makes more sense.